.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,201 @@
+  a breaking change of the API of a package
+- Add a `CLOSES: #<number>` in the footer of a commit to close an issue
+  this commit solves.
+- Commit messages must be written in imperative tense
+- The `scope` defines what package is affected by a commit (enforced by [config-lerna-scopes](https://www.npmjs.com/package/@commitlint/config-lerna-scopes))
+- For changes not associated to a package, scope can be omitted
+
+Relevant types that define changes in [semver](https://semver.org/) versioning are:
+
+- `fix`: Bumps patch version. Use when the commit represents a bugfix
+- `feat`: Bumps minor version. Use when the commit adds a new feature
+
+Other types which do not affect versioning are:
+
+- `build`: Affects the build system or external dependencies.
+- `chore`: Other changes that don't modify src or test files
+- `ci`: Changes CI configuration files and scripts.
+- `docs`: Adds or alters documentation.
+- `perf`: Improves performance.
+- `refactor`: Rewrites code without feature, performance or bug changes.
+- `revert`: Reverts a previous commit.
+- `style`: Improves formatting, white-space.
+- `test`: Adds or modifies tests.
+
+**Tools:**
+
+- If you don't want to worry about formatting, you can run `yarn commit` to
+  run a cli tool that will guide you through the committing process.
+
+** Commit examples: **
+
+Feature commit:
+
+```
+feature(issuer): show modal after transaction finishes
+
+Dispatches an action that opens the modal when transactions finish
+
+CLOSES: #123
+```
+
+Breaking change (bumps MAJOR version):
+
+```
+feat(offchain): migrate to proper rest endpoints
+
+BREAKING CHANGE: Changes all endpoints to return HTTP status codes instead of
+strings inside an object
+
+CLOSES: #123, #456
+```
+
+Commit that doesn't need description, affects multiple packages
+
+```
+style: give a name to all default exported anonymous functions
+```
+
+Some refactor commit:
+
+```
+refactor(ui): cleanup actions in modal
+```
+
+## Branches and Pull Requests
+
+All your contributed code must live in a separate branch before getting merged to the main `develop` branch.
+
+Steps:
+
+- In most cases, you will checkout from `develop` and create a branch
+- Once you're happy with your code, submit a pull request from your branch
+  to `develop`
+- If your PR is out of date, rebase it and update your branch
+
+Rules:
+
+- Rewriting history and therefore force-pushing is **only allowed in personal branches**.
+- No commit reaches `develop` unless it comes from a PR and has been reviewed by at least `1` other developer.
+- No commit reaches `master` except through a PR from `develop` to `master`. _(Note: We might want to ask for more than one approval for this PR since they deploy to production)_
+- You can base your branch from `develop` or an epic branch (e.g: `migration-to-2.0`)
+- PRs that break unit tests will be rejected
+- E2E tests test code in `develop`, `Pull Requests pointing to master` and on `master`
```

Comment:
Shouldn't we mention that the tests results will appear in github, and if the tests break they'll need to be analyzed and the pr put on hold?

Reply:
@josepmc definitely. I'll add this! I'm gonna confirm, but I think I did mention that PRs that break tests are no mergeable at least.